services/heartbeat/types.go

```diff
@@ -23,16 +23,17 @@ package heartbeat
 import (
 	"time"
 
+	"github.com/m3db/m3cluster/services"
 	"github.com/m3db/m3x/watch"
 )
 
 // Store is a heartbeat store
 type Store interface {
 	// Heartbeat sends heartbeat for a service instance with a ttl
-	Heartbeat(service, instance string, ttl time.Duration) error
+	Heartbeat(service string, instance services.PlacementInstance, ttl time.Duration) error
 
 	// Get gets healthy instances for a service
-	Get(service string) ([]string, error)
+	Get(service string) ([]services.PlacementInstance, error)
```

Comment:
I'd consider having 2 different Get(), one gets instance info, the other one gets the ids only (could add WithKeysOnly() in etcd calls) so that use case could be cheaper, also do you think getting keys only is enough for the watch? cause the watch in services interface only cares the instance ids from heartbeat store.